src/commands/Minion/k.ts

```diff
@@ -110,19 +135,75 @@ export default class extends BotCommand {
 				boosts.push(messages.join(' + '));
 			}
 		}
+
 		for (const [itemID, boostAmount] of Object.entries(
 			msg.author.resolveAvailableItemBoosts(monster)
 		)) {
 			timeToFinish *= (100 - boostAmount) / 100;
 			boosts.push(`${boostAmount}% for ${itemNameFromID(parseInt(itemID))}`);
 		}
 
+		// Removed vorkath because he has a special boost.
+		if (
+			monster.name.toLowerCase() !== 'vorkath' &&
+			osjsMon?.data?.attributes?.includes(MonsterAttribute.Dragon)
+		) {
+			if (
+				msg.author.hasItemEquippedOrInBank('Dragon hunter lance') &&
+				!attackStyles.includes(SkillsEnum.Ranged) &&
+				!attackStyles.includes(SkillsEnum.Magic)
+			) {
+				timeToFinish = reduceNumByPercent(timeToFinish, 15);
+				boosts.push('15% for Dragon hunter lance');
+			} else if (
+				msg.author.hasItemEquippedOrInBank('Dragon hunter crossbow') &&
+				attackStyles.includes(SkillsEnum.Ranged)
+			) {
+				timeToFinish = reduceNumByPercent(timeToFinish, 15);
+				boosts.push('15% for Dragon hunter crossbow');
+			}
+		}
+		// Add 15% slayer boost on task if they have black mask or similar
+		if (attackStyles.includes(SkillsEnum.Ranged) || attackStyles.includes(SkillsEnum.Magic)) {
+			if (isOnTask && msg.author.hasItemEquippedOrInBank(itemID('Black mask (i)'))) {
+				timeToFinish = reduceNumByPercent(timeToFinish, 15);
+				boosts.push('15% for Black mask (i) on non-melee task');
+			}
+		} else if (isOnTask && msg.author.hasItemEquippedOrInBank(itemID('Black mask'))) {
+			timeToFinish = reduceNumByPercent(timeToFinish, 15);
+			boosts.push('15% for Black mask on melee task');
+		}
+
 		const maxTripLength = msg.author.maxTripLength(Activity.MonsterKilling);
 
 		// If no quantity provided, set it to the max.
 		if (quantity === null) {
 			quantity = floor(maxTripLength / timeToFinish);
 		}
+		if (typeof quantity !== 'number') quantity = parseInt(quantity);
+		if (isOnTask) {
+			// Todo: Probably handle this in a separate function to make everything easier.
```

Comment:
All todos should be capitalized `// TODO: <text>`
todos should be addressed in your PR if possible, although not totally required.